nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DebugFlow.java

```diff
@@ -0,0 +1,414 @@
+            .description("Number of flowfiles to roll back (without penalty).")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+    static final PropertyDescriptor FF_ROLLBACK_YIELD_ITERATIONS = new PropertyDescriptor.Builder()
+            .name("Rollback Yield Iterations")
+            .description("Number of flowfiles to roll back and yield.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+    static final PropertyDescriptor FF_ROLLBACK_PENALTY_ITERATIONS = new PropertyDescriptor.Builder()
+            .name("Rollback Penalty Iterations")
+            .description("Number of flowfiles to roll back with penalty.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+    static final PropertyDescriptor FF_EXCEPTION_ITERATIONS = new PropertyDescriptor.Builder()
+            .name("Exception Iterations")
+            .description("Number of flowfiles to throw NPE exception.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor NO_FF_EXCEPTION_ITERATIONS = new PropertyDescriptor.Builder()
+            .name("No Flowfile Exception Iterations")
+            .description("Number of times to throw NPE exception if no flowfile.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+    static final PropertyDescriptor NO_FF_YIELD_ITERATIONS = new PropertyDescriptor.Builder()
+            .name("No Flowfile Yield Iterations")
+            .description("Number of times to yield if no flowfile.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+    static final PropertyDescriptor NO_FF_SKIP_ITERATIONS = new PropertyDescriptor.Builder()
+            .name("No Flowfile Skip Iterations")
+            .description("Number of times to skip onTrigger if no flowfile.")
+            .required(true)
+            .defaultValue("1")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    Integer FF_SUCCESS_MAX = 0;
+    private Integer FF_FAILURE_MAX = 0;
+    private Integer FF_ROLLBACK_MAX = 0;
+    private Integer FF_YIELD_MAX = 0;
+    private Integer FF_PENALTY_MAX = 0;
+    private Integer FF_EXCEPTION_MAX = 0;
+
+    private Integer NO_FF_EXCEPTION_MAX = 0;
+    private Integer NO_FF_YIELD_MAX = 0;
+    private Integer NO_FF_SKIP_MAX = 0;
+
+    private Integer FF_SUCCESS_CURR = 0;
```

Comment:
as these are not constants, they shouldn't be all caps. also, I believe there will be concurrency issues with this (need locking and safe publication)